src/app/components/elements/panels/EventGenderDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import React from "react";
+import { EventBookingDTO, UserRole } from "../../../../IsaacApiTypes";
+import { Col, Table, UncontrolledTooltip } from "reactstrap";
+import { asPercentage } from "../../../services";
+
+export const countEventDetailsByRole = (role: UserRole | undefined, eventBookings: EventBookingDTO[]) => {
+  const countByRole = {
+    genders: {
+      male: 0,
+      female: 0,
+      other: 0,
+      preferNotToSay: 0,
+      unknown: 0,
+    },
+    numberOfConfirmedOrAttendedBookings: 0,
+  };
+
+  eventBookings.forEach((booking) => {
+    const gender = booking.userBooked?.gender;
+    const bookingStatus = booking.bookingStatus;
+    const userRole = booking.userBooked?.role;
+
+    const roleFilter = role === undefined || userRole === role;
+
+    if (booking.userBooked && bookingStatus && ["CONFIRMED", "ATTENDED"].includes(bookingStatus) && roleFilter) {
+      countByRole.numberOfConfirmedOrAttendedBookings++;
+
+      switch (gender) {
+        case "MALE":
+          countByRole.genders.male++;
+          break;
+        case "FEMALE":
+          countByRole.genders.female++;
+          break;
+        case "OTHER":
+          countByRole.genders.other++;
+          break;
+        case "PREFER_NOT_TO_SAY":
+          countByRole.genders.preferNotToSay++;
+          break;
+        case "UNKNOWN":
+        case undefined:
+          countByRole.genders.unknown++;
+          break;
+        default:
```

Comment:
Shouldn't the default be the same as unknown and undefined? You're already being exhaustive and testing all the different possibilities, so this line is not even reachable currently.

Reply:
Yeah it's probably unnecessary to have an extra scenario for default, since gender can only be one of the defined options.